capplets/appearance/mate-wp-xml.c

```diff
@@ -529,12 +526,17 @@ void mate_wp_xml_save_list(AppearanceData* data)
 		mate_wp_item_free(wpitem);
 	}
 
-	/* Guardamos el archivo, solo si hay nodos en <wallpapers> */
+	/* save the xml document, only if there are nodes in <wallpapers> */
 	if (xmlChildElementCount(root) > 0)
 	{
+		char *wpdir = g_build_path (G_DIR_SEPARATOR_S, g_get_user_config_dir(), "mate", NULL);
+		g_mkdir_with_parents (wpdir, 0700);
+		g_free (wpdir);
+
+		char *wpfile = g_build_filename(g_get_user_config_dir(), "mate", "backgrounds.xml", NULL);
```

Comment:
I think you should reuse `wpdir` here and just add the filename to it instead of duplicating the directory part:
```suggestion
		char *wpfile = g_build_filename (wpdir, "backgrounds.xml", NULL);
```

Reply:
This could lead to conversion problems on some systems.

https://developer.gnome.org/glib/stable/glib-Character-Set-Conversion.html

Reply:
Ok, wip... it looks like g_get_user_config_dir()

Reply:
> This could lead to conversion problems on some systems.

It should not, [`g_build_filename()`](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-build-filename) expect filename encoding arguments, so it should be alright.